docker/recommender_system.py

```diff
@@ -34,6 +34,12 @@ def recommend(user_events, tips, method):
         recommendation = algorithms[method].recommend(user_events, tips)
         logging.info(str(method.name) + " recommender done.")
 
+    recommendation_set = set(recommendation)
+    if len(recommendation) != len(tips):
+        for tip in tips:
```

Comment:
I think It's better to shuffle tips if we are unable to sort them.

Reply:
Done.